staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -742,16 +748,50 @@ func validateCustomResourceDefinitionValidation(ctx context.Context, customResou
 	return allErrs
 }
 
+type costInfo struct {
+	// MaxCardinality tracks the largest number of times a rule in the current schema node could possibly get executed
+	// due to being the child of arrays/maps/etc.
+	MaxCardinality *uint64
```

Comment:
what is the meaning of nil vs *0 vs *1+?

Reply:
I'd probably godoc this as:

```
MaxCardinality represents a limit to the number of data elements that can exist for the current schema based on MaxProperties or MaxItems limits present on parent schemas, If all parent map and array schemas have MaxProperties or MaxItems limits declared MaxCardinality is the int pointer representing the product of these limits.  If least one parent map or list schema does not have a MaxProperties or MaxItems limits set and the MaxCardinality is nil, indicating that the parent schemas offer no bound to the number of times a data element for the current schema can exist"
```